sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuBatchScanExec.scala

```diff
@@ -173,46 +173,94 @@ object GpuCSVScan {
       meta.willNotWorkOnGpu("GpuCSVScan does not support modified escape chars")
     }
 
-    // TODO charToEscapeQuoteEscaping???
+    if (parsedOptions.charToEscapeQuoteEscaping.isDefined) {
+      meta.willNotWorkOnGpu("GPU CSV Parsing does not support charToEscapeQuoteEscaping")
+    }
 
     if (StandardCharsets.UTF_8.name() != parsedOptions.charset &&
         StandardCharsets.US_ASCII.name() != parsedOptions.charset) {
       meta.willNotWorkOnGpu("GpuCSVScan only supports UTF8 encoded data")
     }
 
-    if (parsedOptions.ignoreLeadingWhiteSpaceInRead) {
-      // TODO need to fix this (or at least verify that it is doing the right thing)
-      meta.willNotWorkOnGpu("GpuCSVScan does not support ignoring leading white space")
+    // TODO parsedOptions.ignoreLeadingWhiteSpaceInRead cudf always do this, but not for strings
+    // TODO parsedOptions.ignoreTrailingWhiteSpaceInRead cudf always do this, but not for strings
+    // TODO parsedOptions.multiLine cudf always does this, but it is not the default and it is not
+    //  consistent
+
+    if (parsedOptions.lineSeparator.getOrElse("\n") != "\n") {
+      meta.willNotWorkOnGpu("GpuCSVScan only supports \"\\n\" as a line separator")
     }
 
-    if (parsedOptions.ignoreTrailingWhiteSpaceInRead) {
-      // TODO need to fix this (or at least verify that it is doing the right thing)
-      meta.willNotWorkOnGpu("GpuCSVScan does not support ignoring trailing white space")
+    if (parsedOptions.parseMode != PermissiveMode) {
+      meta.willNotWorkOnGpu("GpuCSVScan only supports Permissive CSV parsing")
     }
 
-    if (parsedOptions.multiLine) {
-      meta.willNotWorkOnGpu("GpuCSVScan does not support multi-line")
+    // TODO parsedOptions.nanValue This is here by default so we should support it, but cudf
+    // make it null https://github.com/NVIDIA/spark-rapids/issues/125
+    parsedOptions.positiveInf.toLowerCase() match {
+      case "inf" | "+inf" | "infinity" | "+infinity" =>
+      case _ =>
+        meta.willNotWorkOnGpu(s"the positive infinity value '${parsedOptions.positiveInf}'" +
+            s" is not supported'")
     }
+    parsedOptions.negativeInf.toLowerCase() match {
+      case "-inf" | "-infinity" =>
+      case _ =>
+        meta.willNotWorkOnGpu(s"the positive infinity value '${parsedOptions.positiveInf}'" +
+            s" is not supported'")
+    }
+    // parsedOptions.maxCharsPerColumn does not impact the final output it is a performance
+    // improvement if you know the maximum size
 
-    if (parsedOptions.lineSeparator.getOrElse("\n") != "\n") {
-      meta.willNotWorkOnGpu("GpuCSVScan only supports \"\\n\" as a line separator")
+    // parsedOptions.maxColumns was originally a performance optimization but is not used any more
+
+    if (readSchema.map(_.dataType).contains(DateType)) {
```

Comment:
Nit: We could make these very-similar-but-not-quite-the-same checks more data-driven and parameterized, e.g.: a sequence of Spark types and conf keys or maybe a trivial case-class to make it easier to write/read the parameters.

Reply:
I'm less convinced about this, especially for DecimalType where contains on the map is not going to work, because a DecimalType is an actual class.  We don't support it yet, but are likely to eventually.